modules/openid_auth/app/controllers/openid_auth/v0/mvi_users_controller.rb

```diff
@@ -55,51 +29,39 @@ def icn_found
             "type": 'user-mvi-icn',
             "data": {
               "attributes": {
-                "icn": @mvi_response.profile.icn
+                "icn": @mvi_response.profile.icn,
+                "first_name": @mvi_response.profile&.given_names&.first,
+                "last_name": @mvi_response.profile&.family_name
               }
             }
           }
       end
 
-      def highest_loa
-        request.headers['x-va-level-of-assurance'].to_i == 3 ? 3 : 1
-      end
-
       def check_required_headers
-        raise Common::Exceptions::ParameterMissing, 'X-VA-SSN or X-VA-EDIPI' if missing_ssn_or_edipi
         raise Common::Exceptions::ParameterMissing, 'x-va-level-of-assurance' if missing_loa
-        raise Common::Exceptions::ParameterMissing, 'x-va-user-email' if missing_email
-      end
-
-      def missing_ssn_or_edipi
-        request.headers['x-va-ssn'].blank? && request.headers['x-va-edipi'].blank?
-      end
-
-      def missing_email
-        request.headers['x-va-user-email'].blank?
       end
 
       def missing_loa
         request.headers['x-va-level-of-assurance'].blank?
       end
 
       def build_identity
-        UserIdentity.create(uuid: ssn_or_edipi,
+        UserIdentity.create(uuid: request.headers['x-va-idp-uuid'],
                             email: request.headers['x-va-user-email'],
                             first_name: request.headers['x-va-first-name'],
                             last_name: request.headers['x-va-last-name'],
+                            # TODO: break this out into a method that handles unknown gender
+                            gender: request.headers['x-va-gender']&.chars&.first&.upcase,
```

Comment:
Is this TODO supposed to be here? If you meant to leave it as a comment for the next person who extends this, maybe change it to FUTURE or CONSIDER?

Reply:
I didn't modify this since I was unsure of how we handle unknown gender. @patrickvinograd what do we need to do for that?

I think the vets-api codebase generally uses TODO in the same way you were suggesting FUTURE or CONSIDER.

Reply:
Interesting. A quick grep agrees. Strange that there's an overcommit config to warn for those. I've never seen TODO used like that before but I suppose I can get onboard.